# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/client/fronts/targets/target-mixin.js
# Commit: 6eaa4d12b58c
# Full Hash: 6eaa4d12b58cb4762729fdf53511450926383db8
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-15 09:40:06
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 10f97f47c883 -r 6eaa4d12b58c devtools/client/fronts/targets/target-mixin.js
--- a/devtools/client/fronts/targets/target-mixin.js	Thu Oct 15 05:11:58 2020 +0000
+++ b/devtools/client/fronts/targets/target-mixin.js	Thu Oct 15 05:17:52 2020 +0000
@@ -411,7 +411,10 @@
     }
 
     get isWorkerTarget() {
-      return this.typeName === "workerDescriptor";
+      // XXX Remove the check on `workerDescriptor` as part of Bug 1667404.
+      return (
+        this.typeName === "workerTarget" || this.typeName === "workerDescriptor"
+      );
     }
 
     get isLegacyAddon() {
@@ -535,7 +538,13 @@
       if (this.isDestroyedOrBeingDestroyed()) {
         return;
       }
-      await this.attach();
+
+      // WorkerTargetFront don't have an attach function as the related console and thread
+      // actors are created right away (from devtools/server/startup/worker.js)
+      if (this.attach) {
+        await this.attach();
+      }
+
       const isBrowserToolbox = targetList.targetFront.isParentProcess;
       const isNonTopLevelFrameTarget =
         !this.isTopLevel && this.targetType === targetList.TYPES.FRAME;